language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionNodeAnalyzer.java

```diff
@@ -357,13 +360,31 @@ public void visit(RecordTypeDescriptorNode node) {
         visitSyntaxNode(node);
     }
 
+    @Override
+    public void visit(BasicLiteralNode node) {
+        if (node.parent().kind() == SyntaxKind.UNARY_EXPRESSION) {
+            node.parent().accept(this);
+        } else {
+            checkAndSetCodeActionNode(node);
+            checkAndSetSyntaxKind(node.kind());
+            visitSyntaxNode(node);
+        }
+    }
```

Comment:
This should be handled inside the code action. Even for a basic literal node with the parent being a unary expression, matchedCodeActionNode should be a basic literal node when highlighted a basic literal node and provide the code action for that node.

Reply:
example: 
```
boolean bool = !true; 
```
here when highlighting `true`

Reply:
Modified the logic to identify the basic literal node in a unary expression. It works as expected for the unary logical expression `!true` and the unary numeric expression `~100`, but doesn't work for the unary numeric expressions `-100` and `+100` since the `typeOf` API returns null for those 2 cases. Created issue https://github.com/ballerina-platform/ballerina-lang/issues/37525 to track that.

Reply:
Yeah observed the same problem hence posted this example with boolean. There are cases like `    boolean boo = !(true && false);` which are invalid, but we still provide the code action. Will make a separate issue and approve this.